pvlib/iotools/solcast.py

```diff
@@ -0,0 +1,443 @@
+    >>>     longitude=151.215297,
+    >>>     output_parameters=['dni', 'clearsky_dni', 'snow_soiling_rooftop'],
+    >>>     api_key="your-key"
+    >>> )
+
+    References
+    ----------
+    .. [1] `Get an API Key <https://toolkit.solcast.com.au/register>`_
+    .. [2] `Solcast API Docs <https://docs.solcast.com.au/>`_
+    """
+
+    params = dict(
+        latitude=latitude,
+        longitude=longitude,
+        format="json",
+        **kwargs
+    )
+
+    data = _get_solcast(
+        endpoint="forecast/radiation_and_weather",
+        params=params,
+        api_key=api_key,
+        map_variables=map_variables
+    )
+
+    return data, {}
+
+
+def get_solcast_live(
+    latitude, longitude, api_key, map_variables=True, **kwargs
+):
+    """Get irradiance and weather estimated actuals for near real-time
+    and past 7 days
+
+    Parameters
+    ----------
+    latitude : float
+        in decimal degrees, between -90 and 90, north is positive
+    longitude : float
+        in decimal degrees, between -180 and 180, east is positive
+    api_key : str
+        To access Solcast data you will need an API key [1]_.
+    map_variables: bool, default: True
+        When true, renames columns of the DataFrame to pvlib variable names
+        where applicable. See variable :const:`VARIABLE_MAP`.
+    kwargs:
+        Optional parameters passed to the GET request
+
+    See [2]_ for full list of parameters.
+
+    Returns
+    -------
+    data : pandas.DataFrame
+        containing the values for the parameters requested.The times
+        in the DataFrame index indicate the midpoint of each interval.
+
+    Examples
+    --------
+    >>> pvlib.iotools.solcast.get_solcast_live(
+    >>>     latitude=-33.856784,
+    >>>     longitude=151.215297,
+    >>>    api_key="your-key"
+    >>> )
+
+    you can pass any of the parameters listed in the API docs, like
+
+    >>> pvlib.iotools.solcast.get_solcast_live(
+    >>>     latitude=-33.856784,
+    >>>     longitude=151.215297,
+    >>>     terrain_shading=True,
+    >>>     output_parameters=['ghi', 'clearsky_ghi', 'snow_soiling_rooftop'],
+    >>>     api_key="your-key"
+    >>> )
+
+    use ``map_variables=False`` to avoid converting the data
+    to PVLib's conventions.
+
+    >>> pvlib.iotools.solcast.get_solcast_live(
+    >>>     latitude=-33.856784,
+    >>>     longitude=151.215297,
+    >>>     map_variables=False,
+    >>>     api_key="your-key"
+    >>> )
+
+    References
+    ----------
+    .. [1] `Get an API Key <https://toolkit.solcast.com.au/register>`_
+    .. [2] `Solcast API Docs <https://docs.solcast.com.au/>`_
+    """
+
+    params = dict(
+        latitude=latitude,
+        longitude=longitude,
+        format="json",
+        **kwargs
+    )
+
+    data = _get_solcast(
+        endpoint="live/radiation_and_weather",
+        params=params,
+        api_key=api_key,
+        map_variables=map_variables
+    )
+
+    return data, {}
+
+
+def solcast2pvlib(data):

```

Comment:
```suggestion
def _solcast2pvlib(data):
```

No need for this function to be public IMHO

Reply:
done.